T5575: ARP/NDP table-size isnt set properly

Apachez- opened the pull request:
<!-- All PR should follow this template to allow a clean and transparent review -->
<!-- Text placed between these delimiters is considered a comment and is not rendered -->

## Change Summary
<!--- Provide a general summary of your changes in the Title above -->
Sysctl values set in /etc/sysctl.d/30-vyos-router.conf incorrectly override values set by vyos-configd during boot.

## Types of changes
<!---
What types of changes does your code introduce? Put an 'x' in all the boxes that apply.
NOTE: Markdown requires no leading or trailing whitespace inside the [ ] for checking
the box, please use [x]
-->
- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes)
- [ ] Migration from an old Vyatta component to vyos-1x, please link to related PR inside obsoleted component
- [ ] Other (please describe):

## Related Task(s)
<!-- All submitted PRs must be linked to a Task on Phabricator. -->
* https://vyos.dev/T5575

## Component(s) name
<!-- A rather incomplete list of components: ethernet, wireguard, bgp, mpls, ldp, l2tp, dhcp ... -->
build, sysctl

## Proposed changes
<!--- Describe your changes in detail -->
Turns out that settings regarding ARP and NDP table-size works when commited during runtime but failes to be properly set during boot.

By removing these lines from /etc/sysctl.d/30-vyos-router.conf the ARP and NDP table-size are now also properly set during boot:

```
# Increase default garbage collection thresholds
net.ipv4.neigh.default.gc_thresh1 = 1024
net.ipv4.neigh.default.gc_thresh2 = 4096
net.ipv4.neigh.default.gc_thresh3 = 8192
#
net.ipv6.neigh.default.gc_thresh1 = 1024
net.ipv6.neigh.default.gc_thresh2 = 4096
net.ipv6.neigh.default.gc_thresh3 = 8192
```

## How to test
<!---
Please describe in detail how you tested your changes. Include details of your testing
environment, and the tests you ran. When pasting configs, logs, shell output, backtraces,
and other large chunks of text, surround this text with triple backtics
```
like this
```
-->
Use this in your VyOS config and commit and save:

```
set system ip arp table-size '32768'
set system ipv6 neighbor table-size '32768'
```

After reboot verify settings with (should read as below):

```
vyos@vyos:~$ sudo sysctl -a | grep -i neigh.default.gc_thr
net.ipv4.neigh.default.gc_thresh1 = 4096
net.ipv4.neigh.default.gc_thresh2 = 16384
net.ipv4.neigh.default.gc_thresh3 = 32768
net.ipv6.neigh.default.gc_thresh1 = 4096
net.ipv6.neigh.default.gc_thresh2 = 16384
net.ipv6.neigh.default.gc_thresh3 = 32768
```

Before the fix the above output (right after boot) was:

```
vyos@vyos:~$ sudo sysctl -a | grep -i neigh.default.gc_thr
net.ipv4.neigh.default.gc_thresh1 = 1024
net.ipv4.neigh.default.gc_thresh2 = 4096
net.ipv4.neigh.default.gc_thresh3 = 8192
net.ipv6.neigh.default.gc_thresh1 = 1024
net.ipv6.neigh.default.gc_thresh2 = 4096
net.ipv6.neigh.default.gc_thresh3 = 8192
```

## Checklist:
<!--- Go over all the following points, and put an `x` in all the boxes that apply. -->
<!--- If you're unsure about any of these, don't hesitate to ask. We're here to help! -->
<!--- The entire development process is outlined here: https://docs.vyos.io/en/latest/contributing/development.html -->
- [x] I have read the [**CONTRIBUTING**](https://github.com/vyos/vyos-1x/blob/current/CONTRIBUTING.md) document
- [x] I have linked this PR to one or more Phabricator Task(s)
- [ ] I have run the components [**SMOKETESTS**](https://github.com/vyos/vyos-1x/tree/current/smoketest/scripts/cli) if applicable
- [x] My commit headlines contain a valid Task id
- [ ] My change requires a change to the documentation
- [ ] I have updated the documentation accordingly

Apachez- committed:
T5575: ARP/NDP table-size isnt set properly

sarthurdev commented:
I think we instead need to see why they are not being overwritten by the config at boot. Files in sysctl.d _should_ be processed by `systemd-sysctl` long before vyos-router is started.

Apachez- commented:
> I think we instead need to see why they are not being overwritten by the config at boot. Files in sysctl.d _should_ be processed by `systemd-sysctl` long before vyos-router is started.

That could be related to the other tasks for example frr vs vyos-router service etc. in terms of execution order during boot.

I assume `/usr/lib/systemd/system/systemd-sysctl.service` is default from Debian so perhaps `/usr/lib/systemd/system/vyos-router.service` should be modified to be dependent on that "systemd-sysctl.service" is done with its stuff before "vyos-router.service" starts doing its stuff?

Commands thats are probably helpful:

```
sudo systemd-analyze dump systemd-sysctl.service
sudo cat /usr/lib/systemd/system/systemd-sysctl.service

sudo systemd-analyze dump vyos-router.service
sudo cat /usr/lib/systemd/system/vyos-router.service

sudo systemd-analyze dump frr.service
sudo cat /usr/lib/systemd/system/frr.service
```

c-po commented:
vyos-router startup script (https://github.com/vyos/vyos-1x/blob/current/src/init/vyos-router#L348) should be extended to call 

```bash
${vyos_conf_scripts_dir}/system-ip.py || log_failure_msg "could not reset system IPv4 optionss"
${vyos_conf_scripts_dir}/system-ipv6.py || log_failure_msg "could not reset system IPv6 options"
```

So proper values and CLI defaults are set during system startup. This should be added to this PR after dropping the values from sysctl file.

Apachez- committed:
T5575: ARP/NDP table-size isnt set properly

Apachez- commented:
> vyos-router startup script (https://github.com/vyos/vyos-1x/blob/current/src/init/vyos-router#L348) should be extended to call
> 
> ```shell
> ${vyos_conf_scripts_dir}/system-ip.py || log_failure_msg "could not reset system IPv4 optionss"
> ${vyos_conf_scripts_dir}/system-ipv6.py || log_failure_msg "could not reset system IPv6 options"
> ```
> 
> So proper values and CLI defaults are set during system startup. This should be added to this PR after dropping the values from sysctl file.

Done!

sarthurdev commented:
@Apachez- have you tested that change adds default and custom values on boot?

Apachez- commented:
> @Apachez- have you tested that change adds default and custom values on boot?

TLDR: Yes :-)

A bit longer:

Verified this on VyOS 1.5-rolling-202309130022.

1. Modified `/etc/sysctl.d/30-vyos-router.conf` according to this commit.

2. Removed these lines from config (to see what default becomes):

```
set system ip arp table-size '32768'
set system ipv6 neighbor table-size '32768'
```

3. Rebooted, result:

```
vyos@vyos:~$ sudo sysctl -a | grep -i neigh.default.gc_thr
net.ipv4.neigh.default.gc_thresh1 = 1024
net.ipv4.neigh.default.gc_thresh2 = 4096
net.ipv4.neigh.default.gc_thresh3 = 8192
net.ipv6.neigh.default.gc_thresh1 = 1024
net.ipv6.neigh.default.gc_thresh2 = 4096
net.ipv6.neigh.default.gc_thresh3 = 8192
```

4. Modified `/usr/libexec/vyos/init/vyos-router` according to this commit.

5. Rebooted, result:

```
vyos@vyos:~$ sudo sysctl -a | grep -i neigh.default.gc_thr
net.ipv4.neigh.default.gc_thresh1 = 1024
net.ipv4.neigh.default.gc_thresh2 = 4096
net.ipv4.neigh.default.gc_thresh3 = 8192
net.ipv6.neigh.default.gc_thresh1 = 1024
net.ipv6.neigh.default.gc_thresh2 = 4096
net.ipv6.neigh.default.gc_thresh3 = 8192
```

6. Added these lines to config (to see what custom becomes):

```
set system ip arp table-size '32768'
set system ipv6 neighbor table-size '32768'
```

7. Before reboot, result:

```
vyos@vyos:~$ sudo sysctl -a | grep -i neigh.default.gc_thr
net.ipv4.neigh.default.gc_thresh1 = 4096
net.ipv4.neigh.default.gc_thresh2 = 16384
net.ipv4.neigh.default.gc_thresh3 = 32768
net.ipv6.neigh.default.gc_thresh1 = 4096
net.ipv6.neigh.default.gc_thresh2 = 16384
net.ipv6.neigh.default.gc_thresh3 = 32768
```

8. After reboot, result:

```
vyos@vyos:~$ sudo sysctl -a | grep -i neigh.default.gc_thr
net.ipv4.neigh.default.gc_thresh1 = 4096
net.ipv4.neigh.default.gc_thresh2 = 16384
net.ipv4.neigh.default.gc_thresh3 = 32768
net.ipv6.neigh.default.gc_thresh1 = 4096
net.ipv6.neigh.default.gc_thresh2 = 16384
net.ipv6.neigh.default.gc_thresh3 = 32768
```

c-po commented:
@MergifyIo backport sagitta
